codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -287,7 +287,7 @@ public ChannelFuture writePushPromise(ChannelHandlerContext ctx, int streamId, i
     @Override
     public ChannelFuture writeGoAway(ChannelHandlerContext ctx, int lastStreamId, long errorCode, ByteBuf debugData,
             ChannelPromise promise) {
-        return lifecycleManager.writeGoAway(ctx, lastStreamId, errorCode, debugData, promise);
+        return lifecycleManager.closeConnectionRemote(ctx, lastStreamId, errorCode, debugData, promise);
```

Comment:
The rational for including `remote` is the local side is free to continue to open streams. The goal was to structure the operations as `[operation][connection | stream][local | remote]` for consistency.  I agree that the `close` terminology may be a bit overloaded...let me see how `goAway` looks...